suite-native/device/src/hooks/useConnectDevice.tsx

```diff
@@ -0,0 +1,58 @@
+import { useEffect } from 'react';
+import { useSelector } from 'react-redux';
+
+import { useNavigation } from '@react-navigation/native';
+
+import { selectDeviceRequestedPin, selectDeviceType } from '@suite-common/wallet-core';
+import { useAlert } from '@suite-native/alerts';
+import {
+    ConnectDeviceStackParamList,
+    ConnectDeviceStackRoutes,
+    RootStackParamList,
+    RootStackRoutes,
+    StackToStackCompositeNavigationProps,
+} from '@suite-native/navigation';
+import TrezorConnect from '@trezor/connect';
+
+type NavigationProps = StackToStackCompositeNavigationProps<
+    ConnectDeviceStackParamList,
+    ConnectDeviceStackRoutes.ConnectDeviceCrossroads,
+    RootStackParamList
+>;
+
+export const useConnectDevice = () => {
+    const hasDeviceRequestedPin = useSelector(selectDeviceRequestedPin);
+
+    const deviceType = useSelector(selectDeviceType);
+    const navigation = useNavigation<NavigationProps>();
+    const { showAlert, hideAlert } = useAlert();
+
+    useEffect(() => {
+        if (hasDeviceRequestedPin) {
+            navigation.navigate(RootStackRoutes.ConnectDevice, {
+                screen: ConnectDeviceStackRoutes.PinMatrix,
+            });
+        }
+    }, [hasDeviceRequestedPin, navigation]);
+
+    // If device is unacquired (restarted app, another app fetched device session, ...),
+    // we cannot work with device anymore. Shouldn't happen on mobile app but just in case.
+    useEffect(() => {
+        if (deviceType === 'unacquired') {
+            showAlert({
+                title: 'We found your connected device in incorrect state.',
+                description: 'Please reconnect your device or ...',
+                icon: 'warningCircle',
+                pictogramVariant: 'red',
+                primaryButtonTitle: 'Steal session',
+                onPressPrimaryButton: () => {
+                    // Get features steals session and starts communication again
+                    TrezorConnect.getFeatures();
+                    hideAlert();
```

Comment:
`hideAlert` is redundant here. It is always called inside the `AlertSheet` right before evaluating the `onPressPrimaryButton` itself.

Reply:
https://github.com/trezor/trezor-suite/pull/9576/commits/a28af4bc7cfec7a3476cfe5c4206fd65afbcdb7f